src/Storages/MergeTree/MergeList.cpp

```diff
@@ -35,13 +58,17 @@ MergeListElement::MergeListElement(const StorageID & table_id_, FutureMergedMuta
     if (!future_part->parts.empty())
     {
         source_data_version = future_part->parts[0]->info.getDataVersion();
-        is_mutation = (result_part_info.getDataVersion() != source_data_version);
+        is_mutation = (result_part_info.level == future_part->parts[0]->info.level) && !is_fake_projection_part;
```

Comment:
Initially, this change has broken `03221_merge_profile_events`:
```
2024-08-20 20:16:53 03221_merge_profile_events:                                             [ FAIL ] 0.57 sec.
2024-08-20 20:16:53 Reason: result differs with reference:  
2024-08-20 20:16:53 --- /repo/tests/queries/0_stateless/03221_merge_profile_events.reference	2024-08-20 20:08:52.769888431 +0200
2024-08-20 20:16:53 +++ /repo/tests/queries/0_stateless/03221_merge_profile_events.stdout	2024-08-20 20:16:53.607374637 +0200
2024-08-20 20:16:53 @@ -1,3 +1,3 @@
2024-08-20 20:16:53  Horizontal	1	20000	3	0	480000	1	1	1	1
2024-08-20 20:16:53  Vertical	1	20000	1	2	480000	1	1	1	1	1	1
2024-08-20 20:16:53 -Vertical	2	400000	2	6	12800000	1	1	1	1	1	1	1	1	1	1
2024-08-20 20:16:53 +Vertical	2	200000	2	6	4800000	1	1	1	1	1	1	1	1	1	1
2024-08-20 20:16:53 
```
That's because projections always have `result_part_info = future_part->parts[0] = {"all", 0, 0, 0}`. So they were considered as mutations, so MutatedUncompressedBytes was incremented instead of MergedUncompressedBytes...